### ISSUE TITLE
Buffer type not working as expected after upgrading from 5.11.7 to 5.11.11

### ISSUE THREAD

**Issue by ultimate-tester** [Automattic/mongoose#9792]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
I have a development machine (mongodb v4.0.5) and a production server (mongodb v4.2.10).
On the development machine, the Buffer type works as expected. I am using Buffers for my users, their hashed password is stored as a buffer in mongodb. Since upgrading to the latest version, the production server however reports that the buffers generated from the official crypto.scrypt method do not match those stored in the database.

**What is the expected behavior?**
Somehow the compatibility with mongodb v4.0.5 is working, but compatibility with v4.2.10 is not. I did see that the way of storing the buffers is different.

_**Development Machine:**_
{ "$binary" : "Z524JJ/hnpJVErIfCgWdCs7Rdkg1q5otM/TkV1XK5QliWBc/7T7f5yzfRTWmW5kcLBs9aFRIpqsSbv+P8DG2Vw==", "$type" : "00" }

_**Production Machine:**_
BinData(0,"ED6wfY+NfEDPXWrdWj+dPr2KbjxsF9idf0ld0PHsb6DIW5OLrVEG3Ec+ys+y5cWKX7ld76/csNAsKnEVPTRtlw==")

**Comment by ultimate-tester** [comment]
I have reverted back to mongoose 5.11.7 which does not fix the issue. The application was working fine yesterday and the only thing I've done in between these days is updating the package.json. I did, however, also execute a db.updateMany manually against the collection in question with a $set operator on different fields. Is it possible this would have updated the fields of buffer type?

NodeJS and MongoDB themselves have not been touched, as proven by the yum history command:
install nodejs           | 2019-10-29 19:18 | Update         |    1 EE
install mongodb-org      | 2019-10-29 18:32 | Install        |    6 EE

**Comment by ultimate-tester** [comment]
I have upgraded mongodb to the same version as production (v4.2.10). The issue did not appear.
Also, I have found out that the BinData vs $binary presentation of the buffer is due to different clients (mongo on production server and Robo 3T on development machine). Using mongo for both presents the data as BinData on both machines.

Now, I have put my production server's data on my development machine and now the issue appears. Somehow these buffers became invalid, I'm betting the updateMany did something funky. Now it's easy to conclude this has nothing to do with mongoose. Should have spent these 30 minutes before I created an issue for it.
